### ISSUE TITLE
Is object returned from RestService.For thread-safe?

### ISSUE THREAD

**Issue by ultimaweapon** [reactiveui/refit#1343]
I guessed it is thread-safe, just want to make sure.

**Comment by glennawatson** [comment]
Yes. The proxy returned from RestService.For<T> (and the AddRefitClient-registered client) is stateless and thread-safe - it holds an HttpClient and an immutable IRequestBuilder, and each call builds its own request. RestService also caches the generated type/factory in static ConcurrentDictionary instances (see src/Refit/RestService.cs). Reuse a single instance across threads; don't create one per call.

Standard HttpClient threading caveats still apply (don't mutate DefaultRequestHeaders concurrently, etc.).

Closing as answered - for more, Discussions is the place: https://github.com/reactiveui/refit/discussions


<blockquote><img src="https://opengraph.githubassets.com/7da521d756b94831ff31d8d5fc75ecbe2c2d69a69928da79595847192e775278/reactiveui/refit" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/reactiveui/refit/discussions">reactiveui refit · Discussions</a></strong></div><div>Explore the GitHub Discussions forum for reactiveui refit. Discuss code, ask questions & collaborate with the developer community.</div></blockquote>
